File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        q=deque()
        q.append((root,1))
        if root is None:
            return 0

        while q:
            cur,lev=q.popleft()

            

            if cur.left is None and cur.right is None:
                return lev
            
            if cur.left :
                q.append((cur.left,lev+1))
            if cur.right:
                q.append((cur.right,lev+1))
            
        return lev

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function (root) {
  if (!root) return 0;

  const queue = [[root, 1]]; 
    while(queue.length>0){
        const [node,depth]=queue.shift()

        //리프 인지 아닌지 확인
        if(!node.left && !node.right){
            return depth
        }
        if(node.left) queue.push([node.left,depth+1])
        if(node.right) queue.push([node.right,depth+1])

    }
    return 0
};